syntax = "proto3";

package machine;

import "google/protobuf/empty.proto";

message Message {
  int32 sender_id = 1;
  int32 logical_clock = 2;
}

service PeerService {
  // Unary call to send a message to a peer
  rpc SendMessage(Message) returns (google.protobuf.Empty);

  // Server-streaming call to receive messages from a peer
  rpc ReceiveMessages(google.protobuf.Empty) returns (stream Message);
}
